pkg/localmetrics/localmetrics.go

```diff
@@ -270,3 +282,71 @@ func (c *MetricsCollector) AddAccountReuseCleanupFailure() {
 func (c *MetricsCollector) SetReconcileDuration(controller string, duration float64) {
 	c.reconcileDuration.WithLabelValues(controller).Observe(duration)
 }
+
+func (c *MetricsCollector) AddAPICall(controller string, req *http.Request, resp *http.Response) {
+	c.apiCallCount.With(prometheus.Labels{
+		"controller": controller,
+		"method":     req.Method,
+		"resource":   resourceFrom(req.URL),
+		"status":     resp.Status,
+	}).Inc()
+}
+
+// resourceFrom normalizes an API request URL, including removing individual namespace and
+// resource names, to yield a string of the form:
+//     $group/$version/$kind[/{NAME}[/...]]
+// or
+//     $group/$version/namespaces/{NAMESPACE}/$kind[/{NAME}[/...]]
+// ...where $foo is variable, {FOO} is actually {FOO}, and [foo] is optional.
+// This is so we can use it as a dimension for the apiCallCount metric, without ending up
+// with separate labels for each {namespace x name}.
+func resourceFrom(url *neturl.URL) (resource string) {
```

Comment:
Generously adapted from [hive](https://github.com/openshift/hive/blob/68fa75297dbfe1a3911bb3e880a9ae0a746d8c75/pkg/controller/utils/clientwrapper.go#L106-L131)

Reply:
This is nice.

Reply:
:)

As noted, stole the concept from hive. I used a slightly different format. But my main contribution was documenting the thing :P